OpenChemistry/tomviz
# Support for merging arrays from different images into a single image

This MR adds to tomviz the ability to select two or more images in the Pipeline view and merge them into an entirely new image in one of two ways. The first way is to merge all arrays from each image into the new image. Arrays with the same name are handled by appending the array's input number to the array name. This is a useful option when the data arrays should maintain some distinction from each other.

The second way to merge image data is to merge all data arrays into a single array by appending all components in the input arrays into a single array in the new image. This can be used to create an RGB image from three separately loaded images.

To test:

* Run tomviz
* Load an image
* Load a second image with the same extent as the first
* Using Ctrl-left mouse button, click on the two loaded images in the Pipeline. It's okay if operators and modules are also selected, only the DataSources will be considered in the selection.
* Right-click on one of the selected DataSources and select **Merge Images**.
* A dialog box will appear that looks like:

![image](https://user-images.githubusercontent.com/360056/30916587-c59042be-a367-11e7-9272-e672863775b1.png)
Choose either option. 
* Click on the new DataSource. The arrays in the new data set will be listed in the **Original Dimensions & Range** and **Transformed Dimensions and Range**.

Caveat: A progress dialog pops up when **Merge array components** is chosen. It does not repaint properly nor does it go away after the operation is complete. I haven't figured out what is going on there yet, hence a WIP.
